server/src/resources/user/mod.rs

```diff
@@ -0,0 +1,201 @@
+                    FieldDefinition::new_internal(PsqlFieldType::TIMESTAMPTZ, false),
+                ),
+                (
+                    "created_at".to_string(),
+                    FieldDefinition::new_internal(PsqlFieldType::TIMESTAMPTZ, true)
+                        .set_default(String::from("CURRENT_TIMESTAMP")),
+                ),
+                (
+                    "updated_at".to_string(),
+                    FieldDefinition::new_internal(PsqlFieldType::TIMESTAMPTZ, true)
+                        .set_default(String::from("CURRENT_TIMESTAMP")),
+                ),
+                (
+                    "deleted_at".to_string(),
+                    FieldDefinition::new_internal(PsqlFieldType::TIMESTAMPTZ, false),
+                ),
+            ]),
+        }
+    }
+
+    /// Converts raw i32 values into string based on matching Enum value
+    fn get_enum_string_val(field: &str, value: i32) -> Option<String> {
+        match field {
+            "auth_method" => AuthMethod::from_i32(value).map(|val| val.as_str_name().to_string()),
+            _ => None,
+        }
+    }
+}
+
+impl GrpcDataObjectType for Data {
+    fn get_field_value(&self, key: &str) -> Result<GrpcField, ArrErr> {
+        match key {
+            "display_name" => Ok(GrpcField::String(self.display_name.clone())),
+            "auth_method" => Ok(GrpcField::I32(self.auth_method)),
+            _ => Err(ArrErr::Error(format!(
+                "Invalid key specified [{}], no such field found",
+                key
+            ))),
+        }
+    }
+}
+
+#[cfg(not(tarpaulin_include))]
+// no_coverage: Can not be tested in unittest until https://github.com/sfackler/rust-postgres/pull/979 has been merged
+impl TryFrom<Row> for Data {
+    type Error = ArrErr;
+
+    fn try_from(row: Row) -> Result<Self, Self::Error> {
+        debug!("Converting Row to user::Data: {:?}", row);
+        Ok(Data {
+            display_name: row.get("display_name"),
+            auth_method: AuthMethod::from_str_name(row.get("auth_method"))
+                .context("Could not convert auth_method column to AuthMethod.")?
+                as i32,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::super::base::test_util::*;
+    use super::*;
+
+    #[test]
+    fn test_user_schema() {
+        let id = Uuid::new_v4().to_string();
+        let data = mock::get_data_obj();
+        let object: ResourceObject<Data> = Object {
+            id,
+            data: Some(data.clone()),
+        }
+        .into();
+        test_schema::<ResourceObject<Data>, Data>(object);
+
+        let result = <ResourceObject<Data> as PsqlType>::validate(&data);
+        assert!(result.is_ok());
+        if let Ok((sql_fields, validation_result)) = result {
+            println!("{:?}", sql_fields);
+            println!("{:?}", validation_result);
+            assert_eq!(validation_result.success, true);
+        }
+    }
+
+    #[test]
+    fn test_scanner_invalid_data() {
```

Comment:
some mentions to `scanner` in here, I think a copy pasta

Reply:
fixed!